packages/app/source/client/controllers/todos-controller.js

```diff
@@ -6,13 +6,17 @@ Space.Object.extend(Todos, 'TodosController', {
   ],
 
   dependencies: {
-    configuration: 'configuration'
+    configuration: 'configuration',
+    todosStore: 'Todos.TodosStore',
```

Comment:
I think the part of the answer is in the event subscription list of the controller. In my case there is no subscription to `TodoCompleted` or `TodoReopened` UI events but there is subscription to `AllTodosToggled` which is maybe specific to `TodoList` component and belongs there. Other components in the UI could maybe use `TodoCompleted` or `TodoReopened` and most probably would not need `AllTodosToggled`... 

Donations are much more complex so the answer is probably there...